### PR TITLE
Integrate with Federated Rails

### PR DESCRIPTION
**Pull request by bradleybeddoes**
TODO:
* Provisioning from federation source
* Highlight in UI when 'researcherness' is detected via
eduPersonEntitlement

Specs are not meant to me exhaustive, but they are meant to explain
the general concepts for interested developers.

### CODE REVIEW

File: app/views/federated_rails/auth/login.html.erb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <div class='content'>
+      <div class="header">Fred Bloggs</div>
+      <div class="meta">Principal: https://idp.one.edu.au/idp/shibboleth!-!d2404817-6fb9-4165-90d8-1</div>
+      <%= form_tag(:controller => 'auth', :action => 'development_login') do %>
+        <%= hidden_field_tag(:principal, 'https://idp.one.edu.au/idp/shibboleth!-!d2404817-6fb9-4165-90d8-1') %>
+        <%= hidden_field_tag(:credential, 'fake-sessionid-webform') %>
+
+        <%= hidden_field :attributes, :display_name, :value => 'Fred Bloggs' %>
+        <%= hidden_field :attributes, :email, :value => 'fredbloggs@one.edu.au' %>
+        <%= hidden_field :attributes, :shared_token, :value => 'SharedTokenForFredBloggs' %>
```
Comment (smangelsdorf) [inline app/views/federated_rails/auth/login.html.erb]: Not a valid shared token

Comment (bradleybeddoes) [inline app/views/federated_rails/auth/login.html.erb]: There is no checks in place for format of ST

Comment (smangelsdorf) [inline app/views/federated_rails/auth/login.html.erb]: Happy days.

**Comment by smangelsdorf** [comment]
Do you care about the Rubocop violations in `federation.rb.dist`? You can explicitly include this file in `.rubocop.yml` if you want, like:

```yaml
AllCops:
  Include:
    - '**/*.rb.dist'
```

Happy to merge as-is though.

**Comment by bradleybeddoes** [comment]
I am actually doing proper federation integration in next PR, will solve that there, good tip however

File: app/views/federated_rails/auth/login.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<h1 class="ui header">Application Login
+  <div class="sub header">Select login account</div>
+</h1>
+
+<div class="ui cards">
+  <div class="card">
```
Comment (smangelsdorf) [inline app/views/federated_rails/auth/login.html.erb]: I love the `ui cards` style for this purpose. Really stylish. :+1:

**Pull request merged**